docs/development/extensions-contrib/spectator-histogram.md

```diff
@@ -0,0 +1,386 @@
+```
+{
+  "type" : "spectatorHistogram",
+  "name" : <output_name>,
+  "fieldName" : <column_name>
+ }
+```
+
+| Property  | Description                                                                                                  | Required? |
+|-----------|--------------------------------------------------------------------------------------------------------------|-----------|
+| type      | This String must be one of "spectatorHistogram", "spectatorHistogramTimer", "spectatorHistogramDistribution" | yes       |
+| name      | A String for the output (result) name of the aggregation.                                                    | yes       |
+| fieldName | A String for the name of the input field containing raw numeric values or pre-aggregated histograms.         | yes       |
+
+### `longSum`, `doubleSum` and `floatSum` aggregators
+To get the population size (count of events contributing to the histogram):
+```
+{
+  "type" : "longSum",
+  "name" : <output_name>,
+  "fieldName" : <column_name_of_aggregated_histogram>
+ }
+```
+
+| Property  | Description                                                                    | Required? |
+|-----------|--------------------------------------------------------------------------------|-----------|
+| type      | Must be "longSum", "doubleSum", or "floatSum".                                 | yes       |
+| name      | A String for the output (result) name of the aggregation.                      | yes       |
+| fieldName | A String for the name of the input field containing pre-aggregated histograms. | yes       |
+
+## Post Aggregators
+
+### Percentile (singular)
+This returns a single percentile calculation based on the distribution of the values in the aggregated histogram.
+
+```
+{
+  "type": "percentileSpectatorHistogram",
+  "name": <output name>,
+  "field": {
+    "type": "fieldAccess",
+    "fieldName": <name of aggregated SpectatorHistogram>
+  },
+  "percentile": <decimal percentile, e.g. 50.0 for median>
+}
+```
+
+| Property   | Description                                                 | Required? |
+|------------|-------------------------------------------------------------|-----------|
+| type       | This String should always be "percentileSpectatorHistogram" | yes       |
+| name       | A String for the output (result) name of the calculation.   | yes       |
+| field      | A field reference pointing to the aggregated histogram.     | yes       |
+| percentile | A single decimal percentile between 0.0 and 100.0           | yes       |
+
+### Percentiles (multiple)
+This returns an array of percentiles corresponding to those requested.
+
+```
+{
+  "type": "percentilesSpectatorHistogram",
+  "name": <output name>,
+  "field": {
+    "type": "fieldAccess",
+    "fieldName": <name of aggregated SpectatorHistogram>
+  },
+  "percentiles": [25, 50, 75, 99.5]
+}
+```
+
+> Note: It's more efficient to request multiple percentiles in a single query
```

Comment:
Where only a single percentile is wanted, often median or 95th, it's slightly nicer to get a single value back, rather than having to extract from an array in the results.
Also not a strong opinion.

Is the note misleading? It's trying to say, "if you want multiple percentiles from the same underlying metric, then ask for them all at once, rather than as separate metrics". 1 query being more efficient than 2.